docs/sources/flow/reference/components/otelcol.processor.attributes.md

```diff
@@ -0,0 +1,582 @@
+        If the key is prefixed with `auth.`, the values are searched
+        in the authentication information set by the server authenticator. 
+        Refer to the server authenticator's documentation part of your pipeline 
+        for more information about which attributes are available.
+        If the key doesn't exist, no action is performed.
+        If the key has multiple values the values will be joined with `;` separator.
+    * `action = "upsert"` is required.
+
+* `hash`: Hashes (SHA1) an existing attribute value.
+
+    * `key` and/or `pattern` is required.
+    * `action = "hash"` is required.
+
+* `extract`: Extracts values using a regular expression rule from the input key to target keys specified in the rule. If a target key already exists, it will be overridden. Note: It behaves similar to the Span Processor `to_attributes` setting with the existing attribute as the source.
+
+    * `key` is required. It specifies the attribute to extract values from. The value of `key` is NOT altered.
+    * `pattern` is required. It is the regex pattern used to extract attributes from the value of `key`. The submatchers must be named. If attributes already exist, they will be overwritten.
+    * `action = "extract"` is required.
+
+* `convert`: Converts an existing attribute to a specified type.
+
+    * `key` is required. It specifies the attribute to act upon.
+    * `action = "convert"` is required.
+    * `converted_type` is required and must be one of int, double or string.
+
+* `delete`: Deletes an attribute from the input data.
+
+    * `key` and/or `pattern` is required. It specifies the attribute to act upon.
+    * `action = "delete"` is required.
+
+### include/exclude blocks
+
+The `include` and `exclude` blocks provide an option to include or exclude data from being fed into the [action] blocks, based on the properties of a span, log, or metric records.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`match_type` | `string` | Controls how items in "services" and "span_names" arrays are interpreted. | | yes
+`services` | `list(string)` | A list of items to match the service name against. | `[]` | no
+`span_names` | `list(string)` | A list of items to match the span name against. | `[]` | no
+`log_bodies` | `list(string)` | A list of strings that the LogRecord's body field must match against. | `[]` | no
+`log_severity_texts` | `list(string)` | A list of strings that the LogRecord's severity text field must match against. | `[]` | no
+`metric_names` | `list(string)` | A list of strings to match the metric name against. | `[]` | no
+`span_kinds` | `list(string)` | A list of items to match the span kind against. | `[]` | no
+
+`match_type` is required and can be set to either `"regexp"` or `"strict"`.
+
+One of the following is also required:
```

Comment:
The Otel docs also mentions [expressions](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/v0.63.0/processor/attributesprocessor#includeexclude-filtering). However, I don't see any "expressions" in the code, so I removed the reference to it here.

Also, the Otel docs didn't mention "span_kinds", which I included here.

I could raise a PR to the Otel repo to also fix it on their end?

Reply:
> I could raise a PR to the Otel repo to also fix it on their end?

Raising it upstream would be good OSS citizen behavior 👍

Reply:
Ok great, I raised a [PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/20855) :)